user-interface/src/main/java/life/qbic/datamanager/views/notifications/toasts/Toast.java

```diff
@@ -0,0 +1,259 @@
+
+/**
+ * A toast notification is a non-modal, unobtrusive window element. It is meant to show information
+ * in brief and auto-expiring windows to the user.
+ * <p>
+ * It is not meant to require user interaction but may provide optional ways for the user to
+ * interact with it.
+ *
+ * @since 1.4.0
+ */
+public final class Toast extends Notification {
+
+  private static final Position DEFAULT_POSITION = Position.BOTTOM_START;
+  private static final boolean DEFAULT_CLOSEABLE = true;
+  private static final boolean DEFAULT_CLOSE_ON_NAVIGATION = true;
+  private static final Level DEFAULT_LEVEL = Level.INFO;
+  private static final int DEFAULT_OPEN_DURATION = 5_000;
+
+  private final List<Registration> closeOnNavigationListeners = new ArrayList<>();
+  private final Button closeButton = closeButton(this);
+
+  private Component content;
+  private Level level;
+  private boolean closeable;
+
+  enum Level {
+    SUCCESS,
+    INFO;
+  }
+
+  static class ToastBuilder {
+
+    Toast build() {
+      return new Toast();
+    }
+
+  }
+
+  private Toast() {
+    super();
+    addClassName("toast-notification");
+    add(closeButton);
+    setCloseable(DEFAULT_CLOSEABLE);
+    setPosition(DEFAULT_POSITION);
+    setType(DEFAULT_LEVEL);
+    closeOnNavigation(DEFAULT_CLOSE_ON_NAVIGATION);
+    setDuration(DEFAULT_OPEN_DURATION);
+  }
+
+  private static Button closeButton(Toast toast) {
+    var button = new Button(LumoIcon.CROSS.create());
+    button.addThemeVariants(ButtonVariant.LUMO_TERTIARY_INLINE);
+    button.addClickListener(it -> toast.close());
+    button.addClassName("close-button");
+    button.setVisible(false);
+    return button;
+  }
+
+  /**
+   * Sets whether toasts are closed after navigation. By default, Toasts no not stay open after
+   * navigation.
+   *
+   * @param close whether to close the toast notification on navigation or not. true - closes the
+   *              toast after navigation.
+   * @return the modified toast
+   */
+  public Toast closeOnNavigation(boolean close) {
+    //reset previous listeners
+    closeOnNavigationListeners.forEach(Registration::remove);
+    closeOnNavigationListeners.clear();
+    if (!close) {
+      return this;
+    }
+    Registration attachListenerRegistration = addAttachListener(closeToastBeforeLeave());
+    closeOnNavigationListeners.add(attachListenerRegistration);
+    return this;
+  }
+
+  private ComponentEventListener<AttachEvent> closeToastBeforeLeave() {
+    return attachEvent -> {
+      BeforeLeaveListener beforeLeaveListener = beforeLeaveEvent -> {
+        if (attachEvent.getSource() instanceof Toast toast) {
+          toast.close();
+        }
+      };
+      UI ui = attachEvent.getUI();
+      Registration beforeLeaveListenerRegistration = ui.addBeforeLeaveListener(beforeLeaveListener);
+      closeOnNavigationListeners.add(beforeLeaveListenerRegistration);
+    };
+  }
+
+  public void setDuration(Duration duration) {
```

Comment:
no need to make it mutable, it can be passed as parameter during construction.

Reply:
The duration is mutable by default. This is given by the Vaadin class. So I adhere to this behaviour by introducing a simple method that takes a duration object instead of milliseconds (int).